### PR TITLE
Add discovered environments

### PR DESCRIPTION
**Pull request by dingmeng-xue**
## Description

Add discovered environments and updated `Add-AzEnvironment` to discover environment settings 

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: src/Accounts/Accounts/Environment/AddAzureRMEnvironment.cs [hunk 1]
```diff
@@ -146,43 +147,70 @@ public string DataLakeAudience
...
 
-        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true,
+        [Parameter(ParameterSetName = EnvironmentPropertiesParameterSet, Mandatory = false, ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource identifier of the Azure Synapse Analytics that is the recipient of the requested token.")]
+        [Parameter(ParameterSetName = MetadataParameterSet, Mandatory = false, ValueFromPipelineByPropertyName = true,
             HelpMessage = "The resource identifier of the Azure Synapse Analytics that is the recipient of the requested token.")]
         public string AzureSynapseAnalyticsEndpointResourceId { get; set; }
 
+        [Parameter(ParameterSetName = DiscoveryParameterSet, Mandatory = true, 
+            HelpMessage = "Discovers environments via default or configured endpoit.")]
```
Comment (isra-fel) [inline src/Accounts/Accounts/Environment/AddAzureRMEnvironment.cs:206]:
```suggestion
            HelpMessage = "Discovers environments via default or configured endpoint.")]
```

File: src/Accounts/Accounts/help/Add-AzEnvironment.md [hunk 2]
```diff
@@ -44,6 +44,12 @@ Add-AzEnvironment [-Name] <String> [[-StorageEndpoint] <String>] [-ARMEndpoint]
  [<CommonParameters>]
 ```
 
+### Discovery
+```
+Add-AzEnvironment [-AutoDiscover] [-Uri <Uri>] [-Scope <ContextModificationScope>]
```
Comment (isra-fel) [inline src/Accounts/Accounts/help/Add-AzEnvironment.md:49]: We should add an example for the parmeter set

Comment (dingmeng-xue) [reply src/Accounts/Accounts/help/Add-AzEnvironment.md:49]: I'd like to provide example once official DNS is enabled. Currently, no official link supports it

**Pull request merged**